### PR TITLE
[RDY] Add a minimap

### PR DESCRIPTION
**Pull request by Unia**
~~CSS still needs to be done properly and~~ I *think* (haven't checked) that the minimap's highlight can go out of view, at least on the right side.

Since I can't, please properly test this on higher-strain samples such as the 100 or 671 strains. ~~The highlight might get too narrow when there are that many ranks.~~

~~Also, we should ask the customer tomorrow if they want it based on the amount of basepairs.~~ ~~EDIT: in the process of converting.~~ Done.

### CODE REVIEW

**Comment by Unia** [comment]
I think this PR should wait until the rewrite of `Strain` is complete.

**Comment by Unia** [comment]
^ Can wait. The changes to be made are minimal. Please review and merge :)

**Comment by pvanagtmaal** [comment]
Looking + working fine. Next reviewer may merge.

File: dnainator-core/src/main/java/nl/tudelft/dnainator/graph/impl/Neo4jGraph.java [hunk 1]
```diff
@@ -133,6 +141,18 @@ public int getMaxRank() {
 	}
 
 	@Override
+	public int getMaxBasePairs() {
+		return query(e -> (int) e.execute(GET_MAX_BASEPAIRS).columnAs("max").next());
+	}
+
+	@Override
+	public int getRankFromBasePair(int base) {
+		Map<String, Object> parameters = new HashMap<>(1);
```
Comment (gfokkema) [inline dnainator-core/src/main/java/nl/tudelft/dnainator/graph/impl/Neo4jGraph.java]: Collections.singletonMap("dist", base) is better.

File: dnainator-javafx/src/main/java/nl/tudelft/dnainator/javafx/drawables/strains/Strain.java [hunk 2]
```diff
@@ -182,4 +188,26 @@ private void loadRank(Integer key, List<Cluster> value) {
...
+	 */
+	public IntegerProperty maxRankProperty() {
+		return maxRank;
+	}
+
+	/**
+	 * Get the rank width.
+	 * @return the rank width.
+	 */
+	public int getRankWidth() {
```
Comment (gfokkema) [inline dnainator-javafx/src/main/java/nl/tudelft/dnainator/javafx/drawables/strains/Strain.java]: This function really shouldn't exist xD.
I know I've used it somewhere as an example, but the rankwidth is not a property we should expose.

Comment (Unia) [inline dnainator-javafx/src/main/java/nl/tudelft/dnainator/javafx/drawables/strains/Strain.java]: The other solution is to move `gotoRank` to `Strain`, but this leaves us with the problem that `Strain` would require a reference to `StrainView` in order to call `setPan`.

Comment (gfokkema) [inline dnainator-javafx/src/main/java/nl/tudelft/dnainator/javafx/drawables/strains/Strain.java]: Fair enough.

**Comment by gfokkema** [comment]
Next person may merge (after Travis passes).

**Comment by Unia** [comment]
Do not merge this yet. There is an issue with the highlight going out of view.

**Comment by Unia** [comment]
Fixed

**Pull request merged**